src/System.Diagnostics.Process/src/System/Diagnostics/ProcessInfo.cs

```diff
@@ -14,19 +15,248 @@ namespace System.Diagnostics
+                else
+                    return _virtualBytes;
+            }
+            set
+            {
+                _virtualBytes = value;
+            }
+        }
+
+        internal long VirtualBytesPeak
+        {
+            get
+            {
+                if (_virtualBytesPeak == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _virtualBytesPeak;
+            }
+            set
+            {
+                _virtualBytesPeak = value;
+            }
+        }
+
+        internal long WorkingSetPeak
+        {
+            get
+            {
+                if (_workingSetPeak == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _workingSetPeak;
+            }
+            set
+            {
+                _workingSetPeak = value;
+            }
+        }
+
+        internal long WorkingSet
+        {
+            get
+            {
+                if (_workingSet == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _workingSet;
+            }
+            set
+            {
+                _workingSet = value;
+            }
+        }
+
+        internal long PageFileBytesPeak
+        {
+            get
+            {
+                if (_pageFileBytesPeak == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _pageFileBytesPeak;
+            }
+            set
+            {
+                _pageFileBytesPeak = value;
+            }
+        }
+
+        internal long PageFileBytes
+        {
+            get
+            {
+                if (_pageFileBytes == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _pageFileBytes;
+            }
+            set
+            {
+                _pageFileBytes = value;
+            }
+        }
+
+        internal long PrivateBytes
+        {
+            get
+            {
+                if (_privateBytes == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _privateBytes;
+            }
+            set
+            {
+                _privateBytes = value;
+            }
+        }
+        
+        internal int SessionId
+        {
+            get
+            {
+                if (_sessionId == -1)
+                    throw new Win32Exception(SR.ProcessInformationUnavailable);
+                else
+                    return _sessionId;
+            }
+            set
+            {
+                _sessionId = value;
+            }
+        }
+
+
+        internal ProcessInfo()
```

Comment:
Could you move this ctor up in the type to be after the fields but before the properties?